sda-commons-server-s3/src/main/java/org/sdase/commons/server/s3/S3Bundle.java

```diff
@@ -111,11 +102,90 @@ public void run(C configuration, Environment environment) throws Exception {
     }
   }
 
+  private S3Client newClient() {
+    return S3Client.builder()
+        .region(getRegion())
+        .endpointOverride(URI.create(s3Configuration.getEndpoint()))
+        .serviceConfiguration(
+            software.amazon.awssdk.services.s3.S3Configuration.builder()
+                .pathStyleAccessEnabled(true)
+                .build())
+        .overrideConfiguration(
+            ClientOverrideConfiguration.builder()
+                .executionInterceptors(List.of(createTracingExecutingInterceptor()))
+                .putAdvancedOption(SdkAdvancedClientOption.SIGNER, createSigner(s3Configuration))
+                .build())
+        .httpClient(UrlConnectionHttpClient.builder().build())
+        .credentialsProvider(
+            createCredentialsProvider(
+                s3Configuration.getAccessKey(), s3Configuration.getSecretKey()))
+        .build();
+  }
+
+  public S3Presigner newPresigner() {
+    return S3Presigner.builder()
+        .s3Client(getClient())
+        .region(getRegion())
+        .credentialsProvider(
+            createCredentialsProvider(
+                s3Configuration.getAccessKey(), s3Configuration.getSecretKey()))
+        .build();
+  }
+
+  /**
+   * @return the region to use for the S3 client, default is 'eu-central-1'
+   */
+  private Region getRegion() {
+    String region = s3Configuration.getRegion();
+    if (region != null && !region.isBlank()) {
+      return Region.of(region);
+    }
+
+    // default
+    return Region.EU_CENTRAL_1;
+  }
+
+  private ExecutionInterceptor createTracingExecutingInterceptor() {
+    // Initialize a telemetry instance if not set.
+    OpenTelemetry currentTelemetryInstance =
+        this.openTelemetry == null ? GlobalOpenTelemetry.get() : this.openTelemetry;
+    return AwsSdkTelemetry.builder(currentTelemetryInstance)
+        .setCaptureExperimentalSpanAttributes(true)
+        .build()
+        .newExecutionInterceptor();
+  }
+
+  private Signer createSigner(S3Configuration s3Configuration) {
+    String signerOverride = s3Configuration.getSignerOverride();
+    if (signerOverride == null || signerOverride.isBlank()) {
+      return AwsS3V4Signer.create();
+    }
+    // I don't know how to do that more elegantly
```

Comment:
Did you have a look at
`SignerFactory#getSignerByTypeAndService(String signerType, final String serviceName)`

Reply:
SignerFactory is part of the v1 SDK. I could not find something similar for v2.
(the testing module still ships v1)